### PR TITLE
feat(compiler): ElementSchema now has explicit DOM schema information

### PR DESCRIPTION
**Pull request by mhevery**
ignore first commit, it is on top of https://github.com/angular/angular/pull/8158

### CODE REVIEW

File: modules/angular2/src/compiler/schema/dom_element_schema_registry.ts [hunk 1]
```diff
@@ -1,28 +1,201 @@
...
-const NAMESPACE_URIS =
-    CONST_EXPR({'xlink': 'http://www.w3.org/1999/xlink', 'svg': 'http://www.w3.org/2000/svg'});
+const EVENT = 'event';
+const BOOLEAN = 'boolean';
+const NUMBER = 'number';
+const STRING = 'string';
+const OBJECT = 'object';
+const PROPERTIES: string[] =
+    CONST_EXPR([
+      '*|%classList,className,id,innerHTML,*beforecopy,*beforecut,*beforepaste,*copy,*cut,*paste,*search,*selectstart,*webkitfullscreenchange,*webkitfullscreenerror,*wheel,outerHTML,#scrollLeft,#scrollTop',
```
Comment (tbosch) [inline modules/angular2/src/compiler/schema/dom_element_schema_registry.ts]: Add schema for the symbols

File: modules/angular2/src/compiler/schema/dom_element_schema_registry.ts [hunk 2]
```diff
@@ -1,28 +1,201 @@
...
+      '@svg:line^^@svg:geometry|',
+      '@svg:linearGradient^^@svg:gradient|',
+      '@svg:mpath^^@svg:|',
+      '@svg:marker^^@svg:|',
+      '@svg:mask^^@svg:|',
+      '@svg:metadata^^@svg:|',
+      '@svg:path^^@svg:geometry|',
+      '@svg:pattern^^@svg:|',
+      '@svg:polygon^^@svg:geometry|',
+      '@svg:polyline^^@svg:geometry|',
```
Comment (tbosch) [inline modules/angular2/src/compiler/schema/dom_element_schema_registry.ts]: Remove `^^`

File: modules/angular2/test/compiler/schema/dom_element_schema_registry_spec.ts [hunk 3]
```diff
@@ -9,24 +9,25 @@ import {
...
-
-  var registry: DomElementSchemaRegistry;
-
-  beforeEach(() => { registry = new DomElementSchemaRegistry(); });
-
   describe('DOMElementSchema', () => {
+    var registry: DomElementSchemaRegistry;
+    beforeEach(() => { registry = new DomElementSchemaRegistry(); });
 
     it('should detect properties on regular elements', () => {
```
Comment (tbosch) [inline modules/angular2/test/compiler/schema/dom_element_schema_registry_spec.ts]: have 1 test for each symbol and inheritance level

File: modules/angular2/src/compiler/schema/dom_element_schema_registry.ts (hunk 1 above)
Comment (mhevery) [inline modules/angular2/src/compiler/schema/dom_element_schema_registry.ts]: added

File: modules/angular2/src/compiler/schema/dom_element_schema_registry.ts (hunk 2 above)
Comment (mhevery) [inline modules/angular2/src/compiler/schema/dom_element_schema_registry.ts]: fixed

File: modules/angular2/test/compiler/schema/dom_element_schema_registry_spec.ts (hunk 3 above)
Comment (mhevery) [inline modules/angular2/test/compiler/schema/dom_element_schema_registry_spec.ts]: added

**Comment by tbosch** [comment]
@rkirov @IgorMinar  What needs to be changed to land this in G3?

**Comment by rkirov** [comment]
Tighten some client code, didn't looks so bad. I fixed part of it and handed off to Hans.

**Comment by mprobst** [comment]
Currently breaks G3, will have to hold off.

**Comment by rkirov** [comment]
Merged with g3. Good to go for master.

**Comment by mprobst** [comment]
@mhevery I believe this is obsolete, closing.

**Pull request closed without merging**